src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
-                                    || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_WEBSEITE
-                                    || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_GEO) {
-                                jp.nextToken();
-                                continue;
-                            }
-                        }
-
-//                        if (DatenFilm.JSON_NAMES[i] ==  DatenFilm.FILM_TITEL) {
-//                            //convert UNICODE et al to java strings.
-//                            datenFilm.arr[DatenFilm.JSON_NAMES[i]] = StringEscapeUtils.unescapeJava(jp.nextTextValue());
-//                        } else 
-                        if (DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_NEU) {
-                            final String value = jp.nextTextValue();
-                            //This value is unused...
-                            //datenFilm.arr[DatenFilm.FILM_NEU_NR] = value;
-                            datenFilm.setNew(Boolean.parseBoolean(value));
-                        } else {
-                            datenFilm.arr[DatenFilm.JSON_NAMES[i]] = jp.nextTextValue();
-                        }
-
-                        /// für die Entwicklungszeit
-                        if (datenFilm.arr[DatenFilm.JSON_NAMES[i]] == null) {
-                            datenFilm.arr[DatenFilm.JSON_NAMES[i]] = "";
+            if (jp.isExpectedStartArrayToken()) {
+                // sind nur die Feldbeschreibungen, brauch mer nicht
+                jp.nextToken();
+                break;
+            }
+        }
+        while (!Config.getStop() && (jsonToken = jp.nextToken()) != null) {
+            if (jsonToken == JsonToken.END_OBJECT) {
+                break;
+            }
+            if (jp.isExpectedStartArrayToken()) {
+                DatenFilm datenFilm = new DatenFilm();
+                for (int i = 0; i < DatenFilm.JSON_NAMES.length; ++i) {
+                    //if we are in FASTAUTO mode, we don´t need film descriptions.
+                    //this should speed up loading on low end devices...
+                    if (workMode == WorkMode.FASTAUTO) {
+                        if (DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_BESCHREIBUNG
+                                || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_WEBSEITE
+                                || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_GEO) {
+                            jp.nextToken();
+                            continue;
                         }
                     }
-                    if (datenFilm.arr[DatenFilm.FILM_SENDER].isEmpty()) {
-                        datenFilm.arr[DatenFilm.FILM_SENDER] = sender;
+                    if (DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_NEU) {
+                        final String value = jp.nextTextValue();
+                        //This value is unused...
+                        //datenFilm.arr[DatenFilm.FILM_NEU_NR] = value;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)